components/workouts/ExerciseList.jsx

```diff
@@ -0,0 +1,154 @@
+import AlphabetSectionList from 'react-native-alphabet-sectionlist';
+import { SearchBar } from 'react-native-elements';
+import ExerciseItem from './ExerciseItem';
+
+const Title = styled.Text`
+  font-family: 'Montserrat_600SemiBold';
+  font-size: 24px;
+  margin: 15px 5%;
+`;
+
+const SectionHeader = styled.Text`
+  font-family: 'Montserrat_600SemiBold';
+  font-size: 20px;
+  padding-left: 2%;
+  background-color: #CAB0FF;
+  color: #EFEFEF;
+
+`;
+
+const ButtonContainer = styled(TouchableOpacity)`
+   background-color: #E2E2E2;
+   border-radius: 20px;
+   padding: 5px;
+   padding-left: 15px;
+   padding-right: 15px;
+   position: absolute;
+   top: 14px;
+   right: 20px;
+`;
+
+const Buttontext = styled.Text`
+  font-family: 'Montserrat_500Medium';
+  font-size: 18px;
+`;
+
+const parseItems = (items) => {
+  // Sort names alphabetically
+  items.sort((a, b) => a.name.localeCompare(b.name));
+
+  // Group by first letter of each name
+  const bucketData = items.reduce((accumulator, item) => {
+    const bucket = item.name[0].toUpperCase();
+
+    // If this is the first time we've seen this letter, create a bucket
+    if (!accumulator[bucket]) {
+      accumulator[bucket] = [item];
+    } else {
+      accumulator[bucket].push(item);
+    }
+
+    return accumulator;
+  }, {});
+  return bucketData;
+};
+
+const renderHeader = ({ section }) => (
+  <SectionHeader>{section.title}</SectionHeader>
+);
+
+const ExerciseList = (props) => {
+  const { items } = props;
+
+  const parsedItems = parseItems(items);
+  const [search, setSearch] = React.useState('');
+  const [filteredDataSource, setFilteredDataSource] = React.useState([]);
+  const [masterDataSource, setMasterDataSource] = React.useState([]);
+  React.useEffect(() => {
+    setFilteredDataSource(parsedItems);
+    setMasterDataSource(parsedItems);
+  }, []);
+  const searchFilterFunction = (text) => {
+    // Check if searched text is not blank
+    if (text) {
+      // Inserted text is not blank
+      // Filter the masterDataSource
+      // Update FilteredDataSource
+      const firstLetter = text[0];
+      if (masterDataSource[firstLetter]) {
+        const newData = masterDataSource[firstLetter].filter((item) => {
+          const itemData = item.name
+            ? item.name.toUpperCase()
+            : ''.toUpperCase();
+          const textData = text.toUpperCase();
+          return itemData.indexOf(textData) > -1;
+        });
+        const finished = {};
+        finished[firstLetter] = newData;
+        setFilteredDataSource(finished);
+        setSearch(text);
+      } else {
+        setFilteredDataSource(masterDataSource);
+        setSearch(text);
+      }
+    } else {
+      // Inserted text is blank
+      // Update FilteredDataSource with masterDataSource
+      setFilteredDataSource(masterDataSource);
+      setSearch(text);
+    }
+  };
+  const [exerciseCount, setExerciseCount] = React.useState(0);
+  const [addedExercises] = React.useState([]);
```

Comment:
Let's move this to the top of your functional component with the other calls to useState